core/src/miner/stratum.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2018 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//! Client-side stratum job dispatcher and mining notifier handler
+
+use std::net::{AddrParseError, SocketAddr};
+use std::sync::Arc;
+
+use cstratum::PushWorkHandler;
+use cstratum::{Error as StratumServiceError, JobDispatcher, Stratum as StratumService};
+use primitives::{Bytes, H256, U256};
+
+use super::super::client::Client;
+use super::super::miner::work_notify::NotifyWork;
+use super::super::miner::{Miner, MinerService};
+
+/// Configures stratum server options.
+#[derive(Debug, PartialEq, Clone)]
+pub struct Config {
+    /// Network address
+    pub listen_addr: String,
+    /// Port
+    pub port: u16,
+    /// Secret for peers
+    pub secret: Option<H256>,
+}
+
+impl Default for Config {
+    fn default() -> Self {
+        Config {
+            listen_addr: "0.0.0.0:0".parse().unwrap(),
+            port: 3333,
+            secret: None,
+        }
+    }
+}
+
+/// Job dispatcher for stratum service
+pub struct StratumJobDispatcher {
+    client: Arc<Client>,
+    miner: Arc<Miner>,
+}
+
+impl JobDispatcher for StratumJobDispatcher {
+    fn initial(&self) -> Option<String> {
+        // initial payload may contain additional data, not in this case
+        self.job()
+    }
+
+    fn job(&self) -> Option<String> {
```

Comment:
It seems this function always return `None`.
What is expected behavior of this function?

Reply:
It is not needed at the moment, so it may be good to delete, but it is needed to operate a minor (reward, ...). 0c69701

Reply:
Could you expand your comment a bit more?
I can't understand what it means that it's safe to delete, but miner wouldn't operate with that.

Reply:
@Kais-DkM I think this commit is the basic of stratum protocol for codechain. so I aim to provide the same degree as getwork and submit JSON-RPC. 

As you know that, stratum protocols for [bitcoin](https://en.bitcoin.it/wiki/Stratum_mining_protocol) and [ethereum](https://pastebin.com/hsjfAW7H) provide additional information in this part to a miner. 

Could you discuss it with me at the developer meeting?

Reply:
Oh, I was confused with `miner` module in codechain.
I guess it's okay for now, but it's worth a discussion. Let's talk about this later again.